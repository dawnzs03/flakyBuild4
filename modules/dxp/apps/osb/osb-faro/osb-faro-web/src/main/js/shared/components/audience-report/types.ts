export type DataPoint = {
	color: string;
	count: number;
	label: string;
};

export type Dataset = {
	data: DataPoint[];
	empty: Empty;
	total: number;
};

export type Empty = {
	message: string;
	show: boolean;
};

export type TData = {
	audienceReport: {
		anonymousUsersCount: number;
		knownUsersCount: number;
		nonsegmentedKnownUsersCount: number;
		segmentedAnonymousUsersCount: number;
		segmentedKnownUsersCount: number;
	};
	segment: {
		metrics: {
			value: string;
			valueKey: string;
		}[];
		total: number;
	};
};

export enum Name {
	Page = 'page',
	Blog = 'blog',
	Form = 'form',
	Journal = 'journal',
	Document = 'document'
}

export interface IAudienceReportBaseCardProps {
	knownIndividualsTitle: string;
	segmentsTitle?: string;
	uniqueVisitorsTitle?: string;
	query: {
		metricName: string;
		name: Name;
	};
}
