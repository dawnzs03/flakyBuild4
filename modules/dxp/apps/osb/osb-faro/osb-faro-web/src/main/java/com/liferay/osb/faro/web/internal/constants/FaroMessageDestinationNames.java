/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Marcos Martins
 */
public class FaroMessageDestinationNames {

	public static final String FARO_CHECK_FARO_PROJECTS_MESSAGE_PROCESSOR =
		"liferay/faro_check_faro_projects_message_processor";

	public static final String FARO_CLEAN_DL_FILE_ENTRY_MESSAGE_PROCESSOR =
		"liferay/faro_clean_dl_file_entry_message_processor";

	public static final String FARO_DAILY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_daily_email_report_message_processor";

	public static final String FARO_MONTHLY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_monthly_email_report_message_processor";

	public static final String FARO_WEEKLY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_weekly_email_report_message_processor";

}