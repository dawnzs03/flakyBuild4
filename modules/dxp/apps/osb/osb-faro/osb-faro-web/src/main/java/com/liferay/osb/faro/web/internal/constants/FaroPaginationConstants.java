/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.util.PropsValues;

import java.util.Map;

/**
 * @author Shinn Lok
 */
public class FaroPaginationConstants {

	public static final int CUR_DEFAULT = 1;

	public static final String ORDER_BY_TYPE_ASC = "asc";

	public static final String ORDER_BY_TYPE_DEFAULT = ORDER_BY_TYPE_ASC;

	public static final String ORDER_BY_TYPE_DESC = "desc";

	public static Map<String, Object> getConstants() {
		return _constants;
	}

	private static final Map<String, Object> _constants =
		HashMapBuilder.<String, Object>put(
			"cur", CUR_DEFAULT
		).put(
			"delta", PropsValues.SEARCH_CONTAINER_PAGE_DEFAULT_DELTA
		).put(
			"deltaValues", PropsValues.SEARCH_CONTAINER_PAGE_DELTA_VALUES
		).put(
			"orderAscending", ORDER_BY_TYPE_ASC
		).put(
			"orderDefault", ORDER_BY_TYPE_DEFAULT
		).put(
			"orderDescending", ORDER_BY_TYPE_DESC
		).build();

}