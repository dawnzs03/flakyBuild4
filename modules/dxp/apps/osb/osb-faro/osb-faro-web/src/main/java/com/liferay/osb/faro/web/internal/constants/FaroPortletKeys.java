/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Shinn Lok
 */
public class FaroPortletKeys {

	public static final String CEREBRO = "cerebro_portlet";

	public static final String FARO = "faro_portlet";

}