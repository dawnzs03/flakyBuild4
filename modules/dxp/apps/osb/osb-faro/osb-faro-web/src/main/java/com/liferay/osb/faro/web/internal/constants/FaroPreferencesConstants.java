/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.portal.kernel.util.HashMapBuilder;

import java.util.Map;

/**
 * @author Matthew Kong
 */
public class FaroPreferencesConstants {

	public static final String SCOPE_GROUP = "group";

	public static final String SCOPE_USER = "user";

	public static Map<String, Object> getScopes() {
		return _scopes;
	}

	private static final Map<String, Object> _scopes =
		HashMapBuilder.<String, Object>put(
			SCOPE_GROUP, SCOPE_GROUP
		).put(
			SCOPE_USER, SCOPE_USER
		).build();

}