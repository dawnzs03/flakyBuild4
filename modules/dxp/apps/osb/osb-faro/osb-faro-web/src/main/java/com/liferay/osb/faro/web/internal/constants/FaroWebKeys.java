/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Matthew Kong
 */
public class FaroWebKeys {

	public static final String CEREBRO_ASSETS_URL = "CEREBRO_ASSETS_URL";

	public static final String CEREBRO_TOUCHPOINTS_URL =
		"CEREBRO_TOUCHPOINTS_URL";

}