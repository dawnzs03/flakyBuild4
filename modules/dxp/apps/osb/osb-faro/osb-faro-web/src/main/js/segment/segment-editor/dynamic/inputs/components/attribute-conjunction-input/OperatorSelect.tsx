import Form from 'shared/components/form';
import React from 'react';
import {Criterion} from '../../../utils/types';
import {DataTypes} from 'event-analysis/utils/types';
import {
	FunctionalOperators,
	RelationalOperators
} from '../../../utils/constants';
import {getDefaultAttributeValue, getOperatorOptions} from './utils';
import {Option, Picker} from '@clayui/core';

interface IOperatorSelectProps {
	dataType: DataTypes;
	onChange: (params: {criterion: Criterion}) => void;
	operatorName: Criterion['operatorName'];
}

const OperatorSelect: React.FC<IOperatorSelectProps> = ({
	dataType,
	onChange,
	operatorName
}) => {
	if (dataType === DataTypes.Boolean) {
		return (
			<Form.GroupItem className='conjunction' label shrink>
				{Liferay.Language.get('is')}
			</Form.GroupItem>
		);
	}

	return (
		<Form.GroupItem shrink>
			<Picker
				className='operator-input'
				items={getOperatorOptions(dataType)}
				onSelectionChange={newOperatorName => {
					let criterion: Criterion = {
						operatorName: newOperatorName as Criterion['operatorName']
					};

					if (
						(newOperatorName === FunctionalOperators.Between &&
							operatorName !== FunctionalOperators.Between) ||
						(newOperatorName !== FunctionalOperators.Between &&
							operatorName === FunctionalOperators.Between)
					) {
						criterion = {
							...criterion,
							value: getDefaultAttributeValue(
								dataType,
								newOperatorName as
									| FunctionalOperators
									| RelationalOperators
							)
						};
					}

					onChange({criterion});
				}}
				selectedKey={operatorName}
			>
				{({label, value}) => <Option key={value}>{label}</Option>}
			</Picker>
		</Form.GroupItem>
	);
};

export default OperatorSelect;
