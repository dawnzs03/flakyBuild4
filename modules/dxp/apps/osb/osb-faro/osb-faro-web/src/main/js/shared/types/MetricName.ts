export enum MetricName {
	Abandonments = 'abandonmentsMetric',
	AnonymousVisitors = 'anonymousVisitorsMetric',
	AvgTimeOnPage = 'avgTimeOnPageMetric',
	BounceRate = 'bounceRateMetric',
	Comments = 'commentsMetric',
	CompletionTime = 'completionTimeMetric',
	Downloads = 'downloadsMetric',
	Entrances = 'entrancesMetric',
	ExitRate = 'exitRateMetric',
	KnownVisitors = 'knownVisitorsMetric',
	Previews = 'previewsMetric',
	Ratings = 'ratingsMetric',
	ReadingTime = 'readingTimeMetric',
	SessionsDuration = 'sessionDurationMetric',
	SessionsPerVisitor = 'sessionsPerVisitorMetric',
	Submissions = 'submissionsMetric',
	Views = 'viewsMetric',
	Visitors = 'visitorsMetric'
}
