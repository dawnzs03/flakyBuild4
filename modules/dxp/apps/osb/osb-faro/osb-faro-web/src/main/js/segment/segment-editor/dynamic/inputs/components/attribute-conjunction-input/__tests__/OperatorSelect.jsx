import OperatorSelect from '../OperatorSelect';
import React from 'react';
import {DataTypes} from 'event-analysis/utils/types';
import {fireEvent, render} from '@testing-library/react';
import {FunctionalOperators} from '../../../../utils/constants';

jest.unmock('react-dom');

describe('OperatorSelect', () => {
	it('should render', () => {
		const {container, getByText} = render(
			<OperatorSelect
				dataType={DataTypes.Number}
				onChange={jest.fn()}
				operatorsName={FunctionalOperators.Between}
			/>
		);
		fireEvent.click(getByText('Select an option'));

		expect(getByText('is greater than')).toBeTruthy();
		expect(getByText('is less than')).toBeTruthy();
		expect(getByText('between')).toBeTruthy();

		expect(container).toMatchSnapshot();
	});
});
