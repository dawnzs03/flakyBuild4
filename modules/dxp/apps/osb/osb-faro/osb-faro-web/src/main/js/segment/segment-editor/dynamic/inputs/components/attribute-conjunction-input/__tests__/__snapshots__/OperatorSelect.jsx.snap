// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorSelect should render 1`] = `
<div>
  <div
    class="form-group-item form-group-item-shrink"
  >
    <button
      aria-activedescendant="gt"
      aria-expanded="true"
      aria-haspopup="listbox"
      aria-owns="clay-id-2"
      class="form-control form-control-select form-control-select-secondary operator-input show"
      role="combobox"
      style=""
      tabindex="0"
      type="button"
    >
      Select an option
    </button>
  </div>
</div>
`;
